nav2_smart_planner/src/smart_planner.cpp

```diff
@@ -201,7 +202,7 @@ SmartPlanner::makePlan(
   wx = goal.position.x;
   wy = goal.position.y;
 
-  if (worldToMap(wx, wy, mx, my)) {
+  if (!worldToMap(wx, wy, mx, my)) {
     if (tolerance <= 0.0) {
```

Comment:
Does the tolerance value actually matter here? Seems we should error out and fail regardless of the tolerance setting here.

Reply:
Carryover code from ROS1, agree, needs to be changed.